### PR TITLE
Chapter3/homework/2

### PR DESCRIPTION
**Pull request by littleNoodle**

### CODE REVIEW

**Comment by littleNoodle** [comment]
Hello，there are still 2 errors anf I have no idea, could you please help to review them.
1st error message:
Expected collection {"Tom,84 0 111 0 109 0 49 0 50 0 51 0Tom,84 0 111 0 109 0 49 0 50 0 51 0"} to contain "Tom,84 0 111 0 109 0 49 0 50 0 51 0".
what does it mean? I don't get how to do and retry some times but all failed.
2nd error message
Expected isLoggedIn to be true, but found False.
 I check it in the program.cs and get the result of True, I don't know why online test result is false.

Thanks in advance~

**Comment by littleNoodle** [comment]
I got the error message meaning,  a newLine bug.
Also after changes, empty file logic is destroyed.
Add more logic.
Not easy process, I need focus more on the details of System.IO

File: Src/BootCamp.Chapter/Credentials.cs [hunk 1]
```diff
@@ -16,7 +21,46 @@ public Credentials(string username, string password)
...
+        {
+            var credit = obj as Credentials;
+            if (credit == null)
+                return false;
+            if (this.Username == credit.Username && this.Password == credit.Password)
+                return true;
             return false;
+            
+        }
+        private void CheckInput(string a,string b)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Credentials.cs:47]: Ambigious name, can't tell what it does by reading it.

File: Src/BootCamp.Chapter/CredentialsManager.cs [hunk 2]
```diff
@@ -1,24 +1,90 @@
...
 
         // TODO: store credentials in credentials file.
         public void Register(Credentials credentials)
         {
-
+            if (!File.Exists(_credentialsFile))
+            {
+                StreamWriter sw = File.AppendText(_credentialsFile);
+                sw.Write(credentials.ToString());
+                sw.Close();
```
Comment (Almantask) [inline Src/BootCamp.Chapter/CredentialsManager.cs:49]: You can wrap this in a using statements.
```
using(var sw = File.AppendText(_credentialsFile))
{
  sw.Write(...);
}
```

File: Src/BootCamp.Chapter/Program.cs [hunk 3]
```diff
@@ -1,9 +1,31 @@
...
         static void Main(string[] args)
         {
+            var credentials = new Credentials("Tom", "Tom123");
+            Console.WriteLine(credentials.Password);
+
+            const string testFile = @"test.txt";
+            const string EmptyFile = @"E:\Users\ke.li.TE\Documents\GitHub\CSharp-From-Zero-To-Hero\Tests\BootCamp.Chapter.Tests\Input\Files\TomTom123Credentials.txt";
+            var credentialsManager = new CredentialsManager(testFile);
+           
+            //var oldContents = File.ReadAllLines(EmptyFile);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Program.cs:18]: Don't forget to cleanup

**Review by Almantask — APPROVED** [review]
Pretty darn good code.

**Pull request closed without merging**